### PR TITLE
Add processing directory argument

### LINKED ISSUE #4: Rethink temporary file handling
Look into using local temp directories (e.g. /tmp) for temp files, then maybe copying them somewhere if they should be kept?

This might mean changing the `cleanup` arguments to some kind of `processing_directory` and/or `keep_processing argument` that if not set defaults to using temp directories and is then deleted?

### PR DESCRIPTION
**Pull request by thomasmfish**
Adds the ability to specify where a the processing files will be stored, which will enable faster processing on some systems, or just neater output for pipelines.

Closes #4

### CODE REVIEW

File: src/sim_recon/files/utils.py [hunk 1]
```diff
@@ -184,3 +187,70 @@ def combine_text_files(
...
+    path: str | PathLike[str] | None,
+) -> Generator[None, None, None]:
+    if path is None:
+        yield None
+        return
+    try_cleanup = not os.path.isdir(path)
+    try:
+        yield None
+    finally:
+        if try_cleanup and os.path.isdir(path) and not os.listdir(path):
```
Comment (PeterC-DLS) [inline src/sim_recon/files/utils.py]: `os.scandir` or [`Path.iterdir`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir) may be better

Comment (thomasmfish) [reply src/sim_recon/files/utils.py]: Good point, I've made the change now

File: src/sim_recon/files/utils.py [hunk 2]
```diff
@@ -184,3 +187,73 @@ def combine_text_files(
...
+    if path is None:
+        yield None
+        return
+    try_cleanup = not os.path.isdir(path)
+    try:
+        yield None
+    finally:
+        if try_cleanup and os.path.isdir(path):
+            with os.scandir(path) as it:
+                directory_empty = all(False for _ in it)  # False if any entries found
```
Comment (PeterC-DLS) [inline src/sim_recon/files/utils.py]: Do you think `not any(it)` is more readable?

**Comment by thomasmfish** [comment]
@PeterC-DLS Are you happy with this PR now?

File: src/sim_recon/files/utils.py (hunk 2 above)
Comment (thomasmfish) [reply src/sim_recon/files/utils.py]: And faster!

**Review by PeterC-DLS — APPROVED** [review]
LGTM

**Pull request merged**